src/scripts/main.js

```diff
@@ -1,3 +1,365 @@
+const rows = field.children;
+const body = document.querySelector('body');
+const score = document.querySelector('.game-score');
+const starMessage = document.querySelector('.message-start');
+const loseMessage = document.querySelector('.message-lose');
+const winMessage = document.querySelector('.message-win');
+let scoreValue = 0;
+let gameBlock = false;
+const arrayIndex = [];
+
+function getRandomInt(min, max) {
+  /* const mn = Math.ceil(min);
+  const mx = Math.floor(max); */
+
+  return Math.floor(Math.random() * (max - min)) + min;
+}
+
+function getStartPoint() {
+  let index;
+  const arrayPoint = [2, 2, 2, 2, 2, 2, 2, 2, 2, 4];
+
+  if (arrayIndex.length >= 10) {
+    arrayIndex.length = 0;
+  }
+
+  function getPoint() {
+    index = getRandomInt(0, 10);
+
+    if (!arrayIndex.includes(index)) {
+      arrayIndex.push(index);
+    } else {
+      getPoint();
+    }
+  }
+
+  getPoint();
+
+  return arrayPoint[index];
+}
+
+function newPoint() {
+  const row = getRandomInt(0, 4);
+  const column = getRandomInt(0, 4);
+
+  if (rows[row].children[column].classList.value === 'field-cell') {
+    const value = getStartPoint();
+    
+    rows[row].children[column].classList
+    = `field-cell field-cell--${value}`;
+    rows[row].children[column].innerText = value;
+  } else {
+    newPoint();
+  }
+}
+
+function lose() {
+  let freeCeil = 16;
+  let xCount = 12;
+  let yCount = 12;
+
+  for (let i = 0; i < 4; i++) {
+    for (let n = 0; n < 4; n++) {
+      if (rows[i].children[n].innerText !== '') {
+        freeCeil--;
+      }
+    }
+  }
+
+  if (freeCeil === 0) {
+    for (let i = 0; i < 3; i++) {
+      for (let n = 0; n < 4; n++) {
+        if (rows[i].children[n].innerText
+          !== rows[i + 1].children[n].innerText) {
+          xCount--;
+        }
+      }
+    }
+  }
+
+  if (freeCeil === 0) {
+    for (let i = 0; i < 4; i++) {
+      for (let n = 0; n < 3; n++) {
+        if (rows[i].children[n].innerText
+          !== rows[i].children[n + 1].innerText) {
+          yCount--;
+        }
+      }
+    }
+  }
+
+  if (freeCeil === 0 && yCount === 0 && xCount === 0) {
+    loseMessage.classList = 'message message-lose';
+    gameBlock = false;
+  }
+}
+
+function checkWin() {
+  if (document.querySelector('.field-cell--2048')) {
+    winMessage.classList = 'message message-win';
+    gameBlock = false;
+  }
+}
+
+start.addEventListener('click', (evnt) => {
+  if (evnt.target.innerText === 'Start') {
+    gameBlock = true;
+    evnt.target.classList = 'button restart';
+    evnt.target.innerText = 'Rerstart';
+    starMessage.classList = 'message message-start hidden';
+    newPoint();
+    newPoint();
+  }
+
+  if (evnt.target.innerText === 'Rerstart') {
+    gameBlock = true;
+    evnt.target.classList = 'button restart';
+    evnt.target.innerText = 'Rerstart';
+    loseMessage.classList = 'message message-lose hidden';
+    winMessage.classList = 'message message-win hidden';
+
+    for (let i = 0; i < 4; i++) {
```

Comment:
Consider using a functional programming method like forEach or reduce to iterate over the arrays.